lib/ruby_ast_gen.rb

```diff
@@ -116,7 +117,25 @@ def self.process_directory(dir_path, output_dir, exclude_regex, max_threads = 10
   end
 
   def self.parse_file(file_path, relative_input_path)
-    code = File.read(file_path)
+    code =
+      if ruby_file?(file_path)
+        File.read(file_path)
+      else
+        file_content = File.read(file_path)
```

Comment:
Could you put the content in the `begin`-`rescue` blocks in a separate function?